src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -15,6 +15,7 @@
 using Stratis.Features.FederatedPeg.Interfaces;
 using Stratis.Features.FederatedPeg.TargetChain;
 
+[assembly: InternalsVisibleTo("Stratis.Bitcoin.Features.FederatedPeg")]
```

Comment:
This feels like a design issue when we need to expose internals like this. Can't we just make the lock public? Or we somehow need to pass the lock to where it is needed.

Reply:
Hmm also what is this namespace? `Stratis.Bitcoin.Features.FederatedPeg` ? 

We use `Stratis.Features.FederatedPeg`

Reply:
Removed. We don't need it.